src/main/java/de/tum/in/www1/artemis/repository/plagiarism/PlagiarismCaseRepository.java

```diff
@@ -79,6 +79,13 @@ public interface PlagiarismCaseRepository extends JpaRepository<PlagiarismCase,
             """)
     List<PlagiarismCase> findByCourseIdAndStudentId(@Param("courseId") Long courseId, @Param("studentId") Long studentId);
 
+    @Query("""
+            SELECT DISTINCT plagiarismCase FROM PlagiarismCase plagiarismCase
+            WHERE plagiarismCase.exercise.id IN :exerciseIds
+            AND plagiarismCase.student.id = :userId
+            """)
+    List<PlagiarismCase> findByStudentIdAndExerciseIds(@Param("userId") Long userId, @Param("exerciseIds") Set<Long> exerciseIds);
```

Comment:
I think providing the course as second parameter would be easier and more straightforward

Reply:
As far as I can tell this would not work because the exerciseIds used to fetch the plagiarism cases do not just correspond to all exercises of the course, so I cannot just hand all courseIds to the PlagiarismCaseRepository and let it retrieve all plagiarism cases where the plagiarism case's exercise has one of those course ids. The order of retrieving the exercises is as follows: `CourseResource: getAllCoursesForDashboard()` -> `List<Course> courses = courseService.findAllActiveWithExercisesAndLecturesAndExamsForUser(user)` -> `CourseService: userVisibleCourses.stream().peek(course -> course.setExercises(exerciseService.filterExercisesForCourse(course, user))` -> `ExerciseService: exercises.stream().filter(Exercise::isVisibleToStudents).collect(Collectors.toSet())`.
Or am I missing something?